### PR TITLE
planner: move `index_usage` code into a separate package

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #46905

Problem Summary: planner: move `index_usage` code into a separate package

### What is changed and how it works?

No logical change, just code movement.

planner: move `index_usage` code into a separate package

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/pingcap/tidb/statistics/handle/index_usage"
```
[the remark this answers is not in the document]
Comment (qw4990) [reply statistics/handle/handle.go]: updated

**Review by hawkingrei — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]
LGTM

**Pull request merged**